pkg/azuredisk/controllerserver.go

```diff
@@ -1219,3 +1241,139 @@ func getEntriesAndNextToken(req *csi.ListSnapshotsRequest, snapshots []compute.S
 
 	return listSnapshotResp, nil
 }
+
+// Creates a new cloud during CreateVolume call to configure the Useragent property of DiskClient
+func SetupNewUserAgentCloud(userAgent string, currentAz *azure.Cloud) (*azure.Cloud, error) {
```

Comment:
Consider `GetCloudProviderWithUserAgent` and move this and supporting functions to same file as `GetCloudProvider`. (This is currently `pkg/azuredisk/azure.go`, but #977 will be moving it to `pkg/azureutils/azure_disk_utils.go`.)

Reply:
pls reuse existing functions, and don't duplicate functions, code duplication would make it difficult to maintain